standalone-metastore/metastore-common/src/test/java/org/apache/hadoop/hive/metastore/utils/RetryTest.java

```diff
@@ -50,8 +56,49 @@ public Void execute() {
     };
     try {
       retriable.run();
+      Assert.fail();
     } catch (Exception e) {
       Assert.assertEquals(RuntimeException.class, e.getClass());
     }
   }
+
+  @Test
+  public void testRetryFailureWithDelay() {
+    Retry<Void> retriable = new Retry<Void>(NullPointerException.class) {
+      @Override
+      public Void execute() {
+        throw new RuntimeException();
+      }
+    };
+    try {
+      retriable.runWithDelay();
+      Assert.fail();
+    } catch (Exception e) {
+      Assert.assertEquals(RuntimeException.class, e.getClass());
```

Comment:
Add one more assertion for the time taken being more than total time for all three attempts( i.e 180 sec)